docs: add documentations for various flags (#1824)
open-policy-agent/gatekeeper: website/docs/customize-startup.md

Gatekeeper's webhook servers undergo a bootstrapping period during which they are unavailable until the initial set of resources (constraints, templates, synced objects, etc...) have been ingested. This prevents Gatekeeper's webhook from validating based on an incomplete set of policies. This wait-for-bootstrapping behavior can be configured.

The `--readiness-retries` flag defines the number of retry attempts allowed for an object (a Constraint, for example) to be successfully added to OPA.  The default is `0`.  A value of `-1` allows for infinite retries, blocking the webhook until all objects have been added to OPA.  This guarantees complete enforcement, but has the potential to indefinitely block the webhook from serving requests.

## Enable profiling using `pprof`

> The `--enable-pprof` flag enables a HTTP server for profiling using the [pprof](https://pkg.go.dev/net/http/pprof) library. By default, it serves to `localhost:6060` but the port can be customized with the `--pprof-port` flag.

Reviewer: ```suggestion
The `--enable-pprof` flag enables an HTTP server for profiling using the [pprof](https://pkg.go.dev/net/http/pprof) library. By default, it serves to `localhost:6060` but the port can be customized with the `--pprof-port` flag.
```

## Disable certificate generation and rotation for Gatekeeper's webhook

> By default, Gatekeepr uses [`open-policy-agent/cert-controller`](https://github.com/open-policy-agent/cert-controller) to handle the webhook's certificate rotation and generation by. If you want to use a third-party solution, you may disable the cert-controller feature using `--disable-cert-rotation`.

Reviewer: ```suggestion
By default, Gatekeeper uses [`open-policy-agent/cert-controller`](https://github.com/open-policy-agent/cert-controller) to handle the webhook's certificate rotation and generation by. If you want to use a third-party solution, you may disable the cert-controller feature using `--disable-cert-rotation`.
```

> By default, Gatekeeper uses [`open-policy-agent/cert-controller`](https://github.com/open-policy-agent/cert-controller) to handle the webhook's certificate rotation and generation by. If you want to use a third-party solution, you may disable the cert-controller feature using `--disable-cert-rotation`.

Reviewer: by? 

```suggestion
By default, Gatekeeper uses [`open-policy-agent/cert-controller`](https://github.com/open-policy-agent/cert-controller) to handle the webhook's certificate rotation and generation. If you want to use a third-party solution, you may disable the cert-controller feature using `--disable-cert-rotation`.
```

## [Alpha] Emit admission and audit events

> The `--emit-admission-events` flag enables the emission of all admission violations as Kubernetes events in the Gatekeeper namespace. This flag in alpha stage and it is set to `false` by default.

Reviewer 2: ```suggestion
The `--emit-admission-events` flag enables the emission of all admission violations as Kubernetes events in the Gatekeeper namespace. This flag is in alpha stage and it is set to `false` by default.
```

The `--emit-audit-events` flag enables the emission of all audit violation as Kubernetes events in the Gatekeeper namespace. This flag in alpha stage and it is set to `false` by default.

> > Note: if the same constraint/audit was violated for more than 10 times in a 10-minute period, the Kubernetes event recorder will aggregate the events, e.g.

Reviewer: can we link to https://github.com/kubernetes/kubernetes/blob/v1.23.3/staging/src/k8s.io/client-go/tools/record/events_cache.go#L429-L438

Reviewer: also add a note about 25 limit

Reviewer: can we add a note about types of different reasons too? FailedAdmission, DryrunViolation, and AuditViolation

> > Note: if the same constraint/audit was violated for more than 10 times in a 10-minute period, the Kubernetes event recorder will aggregate the events, e.g.

Reviewer 2: ```suggestion
> Note: if the same constraint and violating resource tuple was emitted for more than 10 times in a 10-minute period, the Kubernetes event recorder will aggregate the events, e.g.
```

Please refer to https://github.com/open-policy-agent/gatekeeper/pull/739#issue-663468211 for more details on how the event object's 1Reference1 and `Reason` fields are  derived.
